ai_diffusion/ui/widget.py

```diff
@@ -590,25 +593,75 @@ def move_cursor_to_end(self):
             self._single.setCursorPosition(len(self._single.text()))
 
 
+# SpinBox variant that allows manually entering strength values,
+# but snaps to model_steps on step actions (scrolling, arrows, arrow keys).
+class StrengthSpinBox(QSpinBox):
+    min_steps: int | None
+    max_steps: int | None
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        # safe default
+        self.min_steps = None
+        self.max_steps = None
+        # for manual input
+        self.setMinimum(1)
+        self.setMaximum(100)
+
+    def step_info(self, value: int | None = None, pretty: bool = False) -> tuple[int, int] | None:
+        if self.min_steps is None or self.max_steps is None:
+            return None
+        # See workflow.py:_apply_strength
```

Comment:
Better to adapt and call the function directly. The computation might change at some point.

Reply:
done